src/wp-includes/option.php

```diff
@@ -2147,13 +2147,18 @@ function update_network_option( $network_id, $option, $value ) {
 	/*
 	 * If the new and old values are the same, no need to update.
 	 *
-	 * Unserialized values will be adequate in most cases. If the unserialized
-	 * data differs, the (maybe) serialized data is checked to avoid
-	 * unnecessary database calls for otherwise identical object instances.
+	 * An exception applies when no value is set in the database, i.e. the old value is the default.
+	 * In that case, the new value should always be added as it may be intentional to store it rather than relying on the default.
 	 *
-	 * See https://core.trac.wordpress.org/ticket/44956
+	 * See https://core.trac.wordpress.org/ticket/44956 and https://core.trac.wordpress.org/ticket/22192 and https://core.trac.wordpress.org/ticket/59360
 	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	if (
+		$value === $old_value ||
+		(
+			false !== $old_value &&
+			_is_equal_database_value( $old_value, $value )
+		)
+	) {
```

Comment:
We'll need an extra check here to account for `null`.

If `update_network_option()` is called on Single Site, `update_option()` is called. The `option_value` database field is not nullable, so this will be cast to `(string) ""`, which is equal to `false` and `""`. :+1:

However, let's say `update_network_option()` is called on Multisite. The `meta_value` database field is nullable. The current checks will find return true for `null, false`, `null, ''`, and the same if the values are reversed. :-1:

We should only be calling `_is_equal_database_value()` here if:
- We're in Single Site, OR
- We're in Multisite and both `$old_value` and `$value` are **not** `null`.

```suggestion
	if (
		$value === $old_value ||
		(
			false !== $old_value &&
			/*
			 * Multisite uses the `meta_value` database field, which is nullable.
			 * Don't check for values that are equal to `null`.
			 */
			( ! is_multisite() || ( null !== $old_value && null !== $value ) ) &&
			_is_equal_database_value( $old_value, $value )
		)
	) {
```

While we could instead patch `_is_equal_database_value()` to check `is_multisite()`, not all database queries for Multisite will be dealing with `meta_value` or any other nullable field. Making the change here ensures that we're only targeting this context, and not assuming the situation for all other contexts.

Reply:
Thanks @costdev for raising.